### ISSUE TITLE
Question: support ZodDate type

### ISSUE THREAD

**Issue by heivo** [asteasolutions/zod-to-openapi#48]
Hi,

I'm using a preprocessor to cast ISO 8601 date string query params to Date objects.

Example code: `/api/v1/something?fromDate=2022-10-12T06%3A44%3A10Z`

My query definition is something like this:
```ts
z.object({
  fromDate: z
    .preprocess((arg) => {
      if (typeof arg === 'string') {
        const date = new Date(arg);
        if (isValid(date)) {
          return date;
        }
      }
      return arg;
    }, z.date())
    .optional()
    .openapi({
      type: 'string',
      example: new Date(),
      param: { description: 'From date filter, expects an ISO 8601 date string' },
    }),
})
```
It would be nice if the library would automatically handle this type properly so I wouldn't have to set the type field manually. The same also applies to request bodies that could contain serialized date strings.

If this is something you want to support I could contribute a PR, just let me know.

**Comment by heivo** [comment]
Swagger UI renders the above example as
![image](https://user-images.githubusercontent.com/12901858/195272209-6d80e4cd-558a-4050-8e86-b7c5bd808ae9.png)

**Comment by georgyangelov** [comment]
Hey @heivo , that's definitely something we want to support. It's essentially supporting the ZodDate type. Would be great if you can open a PR for that. Some pointers:

- You can add a test in `spec/types/date.spec.ts` similar to the other tests, you can run them with `npm run test`
- The main function that infers types is `OpenAPIGenerator#toOpenAPISchema`, you can add a check for ZodDate there. You will need to add the type to the `isZodType` function definition as well
- Make sure to run `npm run prettier` before submitting the PR - that will format the code according to the repo rules

Let me know if I can assist with anything!

**Comment by AGalabov** [comment]
Fixed in #51
